fe/fe-core/src/main/java/com/starrocks/sql/analyzer/InsertAnalyzer.java

```diff
@@ -282,4 +288,23 @@ public static void analyze(InsertStmt insertStmt, ConnectContext session) {
         insertStmt.setTargetColumns(targetColumns);
         session.getDumpInfo().addTable(database.getFullName(), table);
     }
+
+    private static ExternalOlapTable getOLAPExternalTableMeta(Database database, ExternalOlapTable externalOlapTable) {
+        // copy the table, and release database lock when synchronize table meta
+        ExternalOlapTable copiedTable = new ExternalOlapTable();
+        externalOlapTable.copyOnlyForQuery(copiedTable);
```

Comment:
is this copy safe? do we have the database read lock hold?

Reply:
The analyzer is executed in db read lock